include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -1471,30 +1487,25 @@ void solveRank1ProblemDist(CommSender&& row_comm, CommSender&& col_comm, const S
         // STEP 2c: reduce, then finalize computation with sign and square root (single-thread)
         if (thread_idx == 0) {
           // local reduction from all bulk workers
-          for (SizeType i = 0; i < m_subm_el_lc; ++i) {
+          for (SizeType i = 0; i < m_el_lc; ++i) {
             for (std::size_t tidx = 1; tidx < nthreads; ++tidx) {
               const T* w_partial = ws_cols[tidx]();
               w[i] *= w_partial[i];
             }
           }
 
           tt::sync_wait(ex::when_all(row_comm_chain(),
-                                     ex::just(MPI_PROD, common::make_data(w, m_subm_el_lc))) |
+                                     ex::just(MPI_PROD, common::make_data(w, m_el_lc))) |
                         transformMPI(all_reduce_in_place));
 
+          // TODO check all weights < 0 (!= 0 otherwise q elements are set to zero and then nomr = 0 => nan)
```

Comment:
@rasolca do we want to add an (heavy) assert for this?

Reply:
Yes